### ISSUE TITLE
Changes in `[filepart]` and `[filepath]`

### ISSUE THREAD

**Issue by mike-mccormick** [badeball/cypress-cucumber-preprocessor#748]
### Current behavior

I am unable to use the `[filepart]` or `[filepath]` placeholders to use step definitions when the test and step definition folders are alongside eachother. 

### Desired behavior

Ideally either `[filepart]` would function as it did previously, or a new option would be added that functions like v10, returning: 

* The name of the feature file, minus the extention. 
* The folder above the feature file + the feature file name. 

For example, given the test `cypress/tests/smoke/smoke.feature`, `[filepart]` would yield:

`['smoke', 'smoke/smoke']`

### Test code to reproduce

More details about the issue, and before/after examples can be found on my  [Test Repo](https://github.com/mike-mccormick/cypress-cucumber-prepro-repro).

### Versions

* **Cypress version**: 10.2.0
* **Preprocessor version**:  11.3.0
* **Node version**:  v16.15.1

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
This is a consequence of upgrading to Cypress v10. The `integrationFolder` used to be what paths were resolved relative to. There's nothing left to resolve paths against, other than the project root, because that's all the preprocessor knows about.

You seem to have the idea that `cypress/tests` is your "test root", but it's not really. Tests can be found outside of said directory, you just chose to not put them anywhere else.  Let me ask you this, if I were to put a test file at `v11/cypress/foo.feature`, where would the preprocessor ideally look for step definitions? Do you see how this question reveals the complexity of this?

**Comment by badeball** [comment]
I'm considering introducing a new option mimicking what `integrationFolder` used to do, in order to allow for what many seems to have been wanting lately.

**Comment by mike-mccormick** [comment]
I think mimicking the `integrationFolder` setting in some way would be ideal here. I may have explained myself wrongly, but regardless you're correct that I misunderstood how the path was generated. 

My choice of folder structure is pretty similar to the one enforced by `nonGlobalStepDefinitions` in the predecessor version of the package, it's just (prior to Cypress removing `integrationFolder`) your package makes it significantly easier and cleaner to organise.

**Comment by MDG-JHowley** [comment]
I'd imagine for most use-cases just supporting something akin to `[filename]` would solve the problem. This is the same as @mike-mccormick first suggestion

>The name of the feature file, minus the extension

---

Why get into the complexity of defining a "test root" or integrationFolder and the complexity of resolving a path against it when the user could handle that detail themselves? A common use-case from the old plugin version would be something like this:

```
{
  stepDefinitions: [
    `cypress/definitions/common/*.{js,ts}`, // common
    `cypress/definitions/[filename]/*.{js,ts}`, // feature specific
  ],
  ...
}
```


__

**Comment by badeball** [comment]
@MDjhowley, are you suggesting resolving `[filename]` to `a`, providing the input `cypress/e2e/foo/a.feature`?

**Comment by MDG-JHowley** [comment]
@badeball yes exactly.

**Comment by badeball** [comment]
That would blow up the directory with step definitions for large projects and cause ambiguity issues for colliding file names. Basically a pretty big step back and not something I'm considering.

What I've pretty much settled on in my mind, is to compile the common ancestor paths shared between all spec files and use that as an implicit integration folder, similar to what Cypress already does with some of their [stuff](https://docs.cypress.io/guides/references/migration-guide#Generated-Files). This would revert the changes required when having upgraded to v11.0.0 of the preprocessor, still support everything as before, _and_ allow users to put their hierarchy of step definitions _beside_ the integration folder.

**Comment by badeball** [comment]
I've released v12.0.0-rc1 with the above-mentioned changes. The preliminary changelog can be read [here](https://github.com/badeball/cypress-cucumber-preprocessor/blob/v12/CHANGELOG.md#v1200-rc1). You can install it like shown below.

```
$ npm i @badeball/cypress-cucumber-preprocessor@12.0.0-rc1
```

@mike-mccormick, and anyone else who may be waiting for this, please try out the new version and get back to me with feedback.

**Comment by MDG-JHowley** [comment]
@badeball works for me

**Comment by mike-mccormick** [comment]
I should have it fully tested before the weekend's out (hopefully sooner), thanks Badeball!

**Comment by mike-mccormick** [comment]
@badeball this also works for me!

**Comment by badeball** [comment]
This has been released as v12.0.0.
